### PR TITLE
@ccheever/with custom font

### PR DESCRIPTION
**Pull request by ccheever**
Created an example `with-custom-font` to illustrate how to use custom fonts in Expo.

This is in conjunction with an updated guide in the docs.

### CODE REVIEW

File: with-custom-font/.expo-shared/assets.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (EvanBacon) [inline with-custom-font/.expo-shared/assets.json]: delete this file

File: with-custom-font/.gitignore [hunk 2]
```diff
@@ -0,0 +1,14 @@
+node_modules/**/*
+.expo/*
+npm-debug.*
+*.jks
```
Comment (EvanBacon) [inline with-custom-font/.gitignore]: delete this file

File: with-custom-font/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "printWidth": 100,
+  "tabWidth": 2,
+  "singleQuote": true,
```
Comment (EvanBacon) [inline with-custom-font/.prettierrc]: delete

File: with-custom-font/app.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "expo": {
+    "name": "Blank Template",
+    "slug": "with-custom-font",
+    "privacy": "public",
+    "sdkVersion": "36.0.0",
+    "platforms": [
+      "ios",
+      "android",
+      "web"
+    ],
```
Comment (EvanBacon) [inline with-custom-font/app.json]: delete

File: with-custom-font/app.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    "version": "1.0.0",
+    "orientation": "portrait",
+    "icon": "./assets/icon.png",
+    "splash": {
+      "image": "./assets/splash.png",
```
Comment (EvanBacon) [inline with-custom-font/app.json]: use remote images like in other examples

File: with-custom-font/app.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    },
+    "updates": {
+      "fallbackToCacheTimeout": 0
+    },
+    "assetBundlePatterns": [
+      "**/*"
+    ],
+    "ios": {
+      "supportsTablet": true
+    }
```
Comment (EvanBacon) [inline with-custom-font/app.json]: delete

File: with-custom-font/package.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
+  "main": "node_modules/expo/AppEntry.js",
```
Comment (EvanBacon) [inline with-custom-font/package.json]:
```suggestion

```

File: with-custom-font/package.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  "scripts": {
+    "start": "expo start",
+    "android": "expo start --android",
+    "ios": "expo start --ios",
+    "web": "expo start --web",
+    "eject": "expo eject"
+  },
```
Comment (EvanBacon) [inline with-custom-font/package.json]: delete

File: with-custom-font/package.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    "react": "~16.9.0",
+    "react-dom": "~16.9.0",
+    "react-native": "https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz",
+    "react-native-web": "~0.11.7"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.0.0",
+    "babel-preset-expo": "~8.0.0"
+  },
+  "private": true
```
Comment (EvanBacon) [inline with-custom-font/package.json]:
```suggestion

```

**Comment by EvanBacon** [comment]
Also please create a minimal readme explaining the example: [template](https://github.com/expo/examples/blob/master/contributing.md#format)

**Comment by byCedric** [comment]
Should we tighten up the main gitignore to skip most of these files for future contributors?

File: with-custom-font/App.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+import { useFonts } from '@use-expo/font';
```
Comment (byCedric) [inline with-custom-font/App.js]: Awesome! 🚀

File: with-custom-font/App.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+import { Text, View, StyleSheet } from 'react-native';
+import { AppLoading } from 'expo';
+
+let customFonts = {
+  'Inter-Black': require('./assets/fonts/Inter-Black.otf'),
+  'Inter-SemiBoldItalic': 'https://rsms.me/inter/font-files/Inter-SemiBoldItalic.otf?v=3.12',
```
Comment (byCedric) [inline with-custom-font/App.js]: Do we also want to [showcase the fine-grained control for webfonts](https://docs.expo.io/versions/latest/sdk/font/#fontdisplay)? I recently stumbled on it by accident, could be cool for other people to show off what `expo-font` is capable of? 😄

Comment (EvanBacon) [reply with-custom-font/App.js]: sure let's do it

**Comment by ccheever** [comment]
made all requested changes

**Pull request merged**